### PR TITLE
bpfd: release-0.3.0

### PR DESCRIPTION
**Pull request by astoycos**
All the required changes needed to cut the bpfd  v0.3.0 release.

This includes some RELEASE.md fixes, updating cargo.toml's, adding a new examples deployment config, and updating the changelog.

Additionally it add's one for v0.2.1 which I must have forgotten to do last time.

fixes https://github.com/bpfd-dev/bpfd/issues/764

### CODE REVIEW

**Comment by astoycos** [comment]
This shouldn't change much I just have some things I want to address tomorrow before this merges 👍

File: changelogs/CHANGELOG-v0.3.0.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+The v0.3.0 release is our second official minor release of the the bpfd project :tada
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.0.md:1]: Do we need a disclaimer at the bottom that there were a lot of API changes and we do not claim backwards compatibility?

Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.0.md:1]: :tada is showing up as :tada in the generated markdown.

File: changelogs/CHANGELOG-v0.3.0.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+of if they were loaded by bpfd or another process
+
+- Deprecation of bpfd specific UUIDs for each loaded program in favor of the
+standard generated kernel ID for all programs regardless of what process loaded them.
+
+- Support for some new bpf program types: 
+    * `Uprobe`s
+    * `Uretprobe`s
+    * `Kprobes`s
+    * `KretProbes`s
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.0.md:15]: I think the `s` should be dropped on probe types above. The generated markdown looks weird.

File: changelogs/CHANGELOG-v0.3.0.md (hunk 1 above)
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.0.md:1]: Should we point to the changelog from the website? I didn't realize we had these for the previous releases until I saw this diff.

File: changelogs/CHANGELOG-v0.3.0.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+The v0.3.0 release is our second official minor release of the the bpfd project :tada
+
+The following describes some of the major new features/updates: 
+
+- The ability to list and get kernel information for ALL programs regardless
+of if they were loaded by bpfd or another process
```
Comment (anfredette) [inline changelogs/CHANGELOG-v0.3.0.md:6]:
```suggestion
of whether they were loaded by bpfd or another process
```

File: RELEASE.md [hunk 4]
```diff
@@ -22,11 +22,20 @@ A release for the bpfd project is comprised of the following major components:
   - `quay.io/bpfd/bpfd-operator`
```
Comment (Billy99) [inline RELEASE.md:22]: Should there be a link to this doc in the Development section of the docs Website?

File: changelogs/CHANGELOG-v0.3.0.md (hunk 1 above)
Comment (anfredette) [reply changelogs/CHANGELOG-v0.3.0.md:1]: Sounds like a good idea.

**Review by anfredette — COMMENTED** [review]
Just a couple of nits.  Otherwise, LGTM.

File: changelogs/CHANGELOG-v0.3.0.md (hunk 1 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.0.md:1]: +1 sounds good

Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.0.md:1]: hrm maybe in a follow up? For now they just get automatically put on the release page.

Comment (Billy99) [reply changelogs/CHANGELOG-v0.3.0.md:1]: Sorry, what's "the release page"?

Comment (Billy99) [reply changelogs/CHANGELOG-v0.3.0.md:1]: Nevermind, I just found it. In github the releases. I was looking in the website.

Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.0.md:1]: I'll add it manually after 👍

**Review by Billy99 — APPROVED** [review]
LGTM

File: changelogs/CHANGELOG-v0.3.0.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- The ability to list and get kernel information for ALL programs regardless
+of whether they were loaded by bpfd or another process
+
+- Deprecation of bpfd specific UUIDs for each loaded program in favor of the
+standard generated kernel ID for all programs regardless of what process loaded them.
+
+- Support for some new bpf program types: 
+    * `Uprobe`
+    * `Uretprobe`
+    * `Kprobes`
```
Comment (anfredette) [inline changelogs/CHANGELOG-v0.3.0.md:19]:
```suggestion
    * `Kprobe`
```

File: changelogs/CHANGELOG-v0.3.0.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    * `KretProbes`
```
Comment (anfredette) [inline changelogs/CHANGELOG-v0.3.0.md:20]:
```suggestion
    * `KretProbe`
```

Comment (anfredette) [reply changelogs/CHANGELOG-v0.3.0.md:20]: I just noticed that the kprobe types were plural and teh uprobe types were not.

**Review by anfredette — APPROVED** [review]

**Review by anfredette — APPROVED** [review]
Just one more nit.  Otherwise it looks good to me.

**Comment by anfredette** [comment]
I approved it so you don't have to wait on me, but it would be good if you could make the final change before you make it non-draft and merge it.

File: changelogs/CHANGELOG-v0.3.0.md (hunk 5 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.0.md]: Done!

**Pull request merged**